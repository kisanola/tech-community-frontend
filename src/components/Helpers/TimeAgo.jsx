import React from 'react';

import moment from 'moment';
import 'moment/locale/fr';

export default ({ date, style }) => {
  return <span style={style}>{moment(date).fromNow()}</span>;
};
