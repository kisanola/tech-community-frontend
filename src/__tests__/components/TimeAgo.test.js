import React from 'react';
import TimeAgo from '../../components/Helpers/TimeAgo';
import { mount } from 'enzyme';

describe('TimeAgo', () => {
  let wrapper;
  const date = new Date().getTime();

  beforeAll(() => {
    wrapper = mount(<TimeAgo date={date} />);
  });

  it('should return the date in `time ago` format', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toBe('il y a quelques secondes');
  });
});
