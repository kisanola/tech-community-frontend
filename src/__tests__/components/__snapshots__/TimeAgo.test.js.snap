// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeAgo should return the date in \`time ago\` format 1`] = `ReactWrapper {}`;
